pkg/auth/providers/common/ldap/ldap_util.go

```diff
@@ -312,3 +323,25 @@ func NewDefaultSearchRequest(baseDN, filter string, scope int, attributes []stri
 		nil,                      // Controls
 	)
 }
+
+// According to RFC4512, attribute names (descriptors) should adhere to the following syntax
+// descr = keystring
+// keystring = leadkeychar *keychar
+// leadkeychar = ALPHA
+// keychar = ALPHA / DIGIT / HYPHEN
+// See https://datatracker.ietf.org/doc/html/rfc4512#section-1.4
+var (
+	numHyphenPrefix   = regexp.MustCompile(`^[0-9\-]`)
+	notAlphaNumHyphen = regexp.MustCompile(`[^a-zA-Z0-9\-]`)
+	validAttr         = regexp.MustCompile(`^[a-zA-Z][a-zA-Z0-9\-]*$`)
+)
+
+// SanitizeAttr removes all invalid character from the LDAP attribute name.
+func SanitizeAttr(attr string) string {
+	return numHyphenPrefix.ReplaceAllString(notAlphaNumHyphen.ReplaceAllString(attr, ""), "")
```

Comment:
What do you think about warning that we've sanitized the annotation?

Reply:
I'm not strongly opposed to the idea, but IMHO it's akin to escaping/sanitazing values for which normally log entries are not produced.